demo/src/Core/BottomTabBar/index.tsx

```diff
@@ -13,43 +11,37 @@ import type { Props } from './types';
  */
 export const BottomTabBar = ({ id }: Props): JSX.Element | null => {
   // id is provided by Hyperview, and represents a tab navigator id
-  const ctx = useContext(BottomTabBarContext);
+  const { getElementProps, setElement } = useBottomTabBarContext();
 
   // Props are the props received by the component backing the custom
   // Hyperview element <navigation:bottom-tab-bar>
-  const props = ctx.elementsProps?.[id];
+  const props = getElementProps?.(id);
+  const { onUpdate } = props || {};
 
-  // It is assumed here the component will nest a single child
-  const child = Array.from(props?.element?.childNodes || []).find(
-    node => node.nodeType === NODE_TYPE.ELEMENT_NODE,
-  ) as Element | undefined;
-
-  // Since the state of the element is no longer held by the screen's doc
-  // that provides it, we create a local state to store the child element
-  // and keep it in sync with the child provided by the props
-  const [element, setElement] = useState<Element | undefined>(child);
-  useEffect(() => {
-    setElement(child);
-  }, [child]);
-  if (!props || !child) {
+  const onUpdateCustom = useCallback(
+    (href, action, currentElement, opts) => {
+      if (action === 'swap' && opts?.newElement) {
+        if (currentElement.parentNode) {
+          const newElement = currentElement.parentNode as Element;
+          newElement.replaceChild(opts.newElement, currentElement);
+          setElement?.(id, newElement);
+        } else {
+          console.warn('Parent node is null. Cannot replace child element.');
```

Comment:
Better to use Logging.warn. See implementations in places like https://github.com/Instawork/hyperview/blob/231688b0b8416c74f2afa57b0edfffa2dfebfd33/src/behaviors/hv-unselect-all/index.ts

I found other console calls in the new tab bar components which aren't part of this PR.
```suggestion
          console.warn('Parent node is null. Cannot replace child element.');
```

Reply:
I don't believe the demo app has the logger set up already?

Reply:
Disregard. This is not internal to hyperview.